### ISSUE TITLE
Latest update of intellij 2017.3 and the dart plugin throw errors consistently

### ISSUE THREAD

**Issue by alexeieleusis** [dart-lang/sdk#31510]
Saw this in mac os and linux

Analyzer Feedback from IntelliJ

# Version information

- `IDEA IC-173.3727.127`
- `1.24.2`
- `IC-173.3727.127, JRE 1.8.0_152-release-1024-b6x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.1 unknown, screens 1920x1200  Retina`

# Exception

Please append the contents of:
file:///private/var/folders/0_/f5tft21n2551p9xq9h8tcvfh007ms_/T/report529.txt

```
Dart analysis server, SDK version 1.24.2, server version 1.18.0, error: Analysis failed: removed file name
exception_20171201_094241_699
UnimplementedError
#0      CompilationUnitElementForLink.resolveTypeRef (package:analyzer/src/summary/link.dart:1096)
#1      CompilationUnitElementForLink.resolveTypeRef.getTypeArgument (package:analyzer/src/summary/link.dart:1103)
#2      new List.generate (dart:core/list.dart:142)
#3      ClassElementForLink_Class.buildType (package:analyzer/src/summary/link.dart:647)
#4      CompilationUnitElementForLink.resolveTypeRef (package:analyzer/src/summary/link.dart:1113)
#5      VariableElementForLink.declaredType (package:analyzer/src/summary/link.dart:5196)
#6      VariableElementForLink.type (package:analyzer/src/summary/link.dart:5287)
#7      PropertyAccessorElementForLink_Variable.returnType (package:analyzer/src/summary/link.dart:4548)
#8      PropertyAccessorElementForLink_Variable.asStaticType (package:analyzer/src/summary/link.dart:4501)
#9      ExprTypeComputer._doPushReference (package:analyzer/src/summary/link.dart:2485)
#10     ExprTypeComputer.compute (package:analyzer/src/summary/link.dart:2073)
#11     TypeInferenceNode.evaluate (package:analyzer/src/summary/link.dart:4957)
#12     TypeInferenceDependencyWalker.evaluate (package:analyzer/src/summary/link.dart:4808)

**Comment by MichaelRFairhurst** [comment]
Hey Alexei!

This is a pretty old SDK, can you try with the latest?

**Comment by alexeieleusis** [comment]
I brew updated yesterday! Even uninstalled and reinstalled, what changed?

**Comment by MichaelRFairhurst** [comment]
Unfortunately due to dart 2, the latest stable is really far behind.

I think this commit fixed it: https://codereview.chromium.org/2931123002 but if you can try to repro on a 2.0.0-dev release we can confirm.

Based on the description there, it looks like this is caused by using generic function syntax (`T Function<T, Z>(T, Z)`) which still has some issues today (ie https://github.com/dart-lang/sdk/issues/30761) so retesting may be a good idea. I think maybe a dependent library doesn't have the right SDK constraint, and therefore maybe pub gave you something that would crash?

I'm not sure how to update to the latest on a mac though...

...do you have the code to repro? I can easily try it against bleeding edge on my system.

**Comment by alexeieleusis** [comment]
I will try to clean it up, is mixed with a bunch of things that won't be public, but your explanation gives me a good idea where to look at, it might be a dependency I wrote myself. Thanks!

I guess that kind of code is not that common though.

**Comment by xxgreg** [comment]
I also ran into this. Upgrading to 2.0-dev got rid of the errors.

Looks like a lot of people are hitting this:
https://github.com/dart-lang/sdk/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+resolveTypeRef

**Comment by bwilkerson** [comment]
I'm guessing that this issue has been resolved in the latest 2.0-dev build. If not, please re-open it.
